raft/src/main/java/org/apache/kafka/snapshot/FileRawSnapshotWriter.java

```diff
@@ -53,13 +54,13 @@ public long sizeInBytes() throws IOException {
     }
 
     @Override
-    public void append(ByteBuffer buffer) throws IOException {
+    public void append(UnalignedMemoryRecords records) throws IOException {
```

Comment:
It might be helpful to have an `append(MemoryRecords)`. That saves an annoying conversion when we are generating the snapshot instead of copying it from the leader.